x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -46,9 +54,21 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       : TimeoutErrorMode.CONTACT;
   }
 
+  private createRequestHash$(request: IKibanaSearchRequest, options: IAsyncSearchOptions) {
+    const { sessionId } = options;
+    const { preference, ...params } = request.params || {};
+    const hashOptions = {
+      ...params,
+      sessionId,
+    };
+
+    return from(sessionId ? createRequestHash(hashOptions) : of(undefined));
```

Comment:
Curious if this is really necessary? 
Why wouldn't we allow cache requests that do not use `sessionId`?